src/Classes/ArchivesListControl.lua

```diff
@@ -0,0 +1,502 @@
+	if not self:IsShown() or not self:IsEnabled() then
+		return
+	end
+	local mOverControl = self:GetMouseOverControl()
+	if mOverControl and mOverControl.OnKeyDown then
+		return mOverControl:OnKeyDown(key)
+	end
+	if not self:IsMouseOver() and key:match("BUTTON") then
+		return
+	end
+end
+
+function ArchivesListClass:OnKeyUp(key)
+	if not self:IsShown() or not self:IsEnabled() then
+		return
+	end
+
+	if self.controls.scrollBarV:IsScrollDownKey(key) then
+		self.controls.scrollBarV:Scroll(1)
+	elseif self.controls.scrollBarV:IsScrollUpKey(key) then
+		self.controls.scrollBarV:Scroll(-1)
+	end
+
+	if key == "LEFTBUTTON" then
+		self:CheckButtons()
+	end
+end
+
+function ArchivesListClass:GetHoveredButton()
+	if self.inTransition then
+		return
+	end
+	local cursorX, cursorY = GetCursorPos();
+	cursorY = cursorY + self.controls.scrollBarV.offset
+	for i, importButton in ipairs(self.importButtons) do
+		if (cursorX > importButton.x0) and
+			(cursorX < importButton.x1) and
+			(cursorY > importButton.y0) and
+			(cursorY < importButton.y1) then
+			return {
+				button = importButton,
+				type = 'import'
+			}
+		end
+	end
+
+	for i, previewButton in ipairs(self.previewButtons) do
+		if (cursorX > previewButton.x0) and
+			(cursorX < previewButton.x1) and
+			(cursorY > previewButton.y0) and
+			(cursorY < previewButton.y1) then
+			return {
+				button = previewButton,
+				type = 'preview'
+			}
+		end
+	end
+end
+
+function ArchivesListClass:GetApiUrl()
+	local archivesUrl = 'https://pobarchives.com'
+	local apiPath = '/api/builds'
+	return archivesUrl .. apiPath .. '?q=' .. self.mode
+end
+
+function ArchivesListClass:GetPageUrl()
+	local archivesUrl = 'https://pobarchives.com'
+	local buildsPath = '/builds'
+	if self.mode == "latest" then
+		return archivesUrl .. buildsPath .. '/yenTGNDb'
+	end
+	if self.mode == "trending" then
+		return archivesUrl .. buildsPath .. '/7U8QXU8m?sort=popularity'
+	end
+	if self.mode == "similar" then
+		return archivesUrl .. buildsPath .. '/?similar=' .. self.similarTo
+	end
+
+	return nil
+end
+
+function ArchivesListClass:GetAscendancyImageHandle(ascendancy)
+	local image = nil
+	if ascendancy then
+		image = NewImageHandle()
+		image:Load(s_format('Assets/ascendants/%s.jpeg', (ascendancy:gsub("^%l", string.lower))))
+	end
+
+	return image
+end
+
+function ArchivesListClass:HandleButtonClick(button, buttonType)
+	if button then
+		self.inTransition = true
+		if buttonType == "import" then
+			local urlText = button.build_link:gsub("^[%s?]+", ""):gsub("[%s?]+$", "") -- Quick Trim
+			local websiteInfo = nil
+			for j = 1, #buildSites.websiteList do
+				if urlText:match(buildSites.websiteList[j].matchURL) then
+					websiteInfo = buildSites.websiteList[j]
+				end
+			end
+
+			if websiteInfo then
+				buildSites.DownloadBuild(button.build_link, websiteInfo, function(isSuccess, data)
+					if isSuccess then
+						local xmlText = Inflate(common.base64.decode(data:gsub("-", "+"):gsub("_", "/")))
```

Comment:
Sidenote: this might be nice to extract to an import library of sorts

Reply:
Moved this to a function under Common. Also tried to change the usage in the ImportTab but it's just too custom to use a generic function so I left that class as is.